starter-code/src/viking.js

```diff
@@ -1,11 +1,97 @@
 // Soldier
-class Soldier {}
+class Soldier {
+    constructor(health, strength) {
+        this.health = health
+        this.strength = strength
+    }
+
+    attack(){return this.strength}
+
+    receiveDamage(damage){
+        this.health -= damage
+    }
+}
+
+/*A Viking is a Soldier with an additional property, their name. They also have a different receiveDamage() method and new method,
+ battleCry().
+Modify the Viking class, have it inherit from Soldier, re-implement the receiveDamage() method for Viking, and add a new battleCry() method.
+*/
 
 // Viking
-class Viking {}
+class Viking extends Soldier{
+    constructor(name, health, strength){
+        super(health, strength)             // always call super first 
+        this.name = name
+    }
+
+
+    receiveDamage(damage){
+        this.health -= damage
+        
+        if(this.health > 0){return `${this.name} has received ${damage} points of damage`}
+
+        if(this.health <= 0){return `${this.name} has died in act of combat`}
+    }
+
+    battleCry(){
+        return "Odin Owns You All!"
+    }
+}
 
 // Saxon
-class Saxon {}
+class Saxon extends Soldier{
+
+    receiveDamage(damage){
+        this.health -= damage
+
+        if(this.health > 0){return `A Saxon has received ${damage} points of damage`}
+
+        if(this.health <= 0){return `A Saxon has died in combat`}
+    }
+}
+
 
 // War
-class War {}
+class War {
+    constructor(){
+        this.vikingArmy = []
+        this.saxonArmy = []
+    }
+
+    addViking(viking){
+       this.vikingArmy.push(viking)
+    }
+
+    addSaxon(saxon){
+        this.saxonArmy.push(saxon)
+    }
+
+    vikingAttack(){
+     let randomSaxonPosition = Math.floor((Math.random() * this.saxonArmy.length))
+     let randomVikingPosition = Math.floor((Math.random() * this.vikingArmy.length))
+
+        this.saxonArmy[randomSaxonPosition].receiveDamage(this.vikingArmy[randomVikingPosition].strength)
```

Comment:
here you should write 
let result = this.saxonArmy[randomSaxonPosition].receiveDamage(this.vikingArmy[randomVikingPosition].strength)